Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -15,7 +20,19 @@ class APIClient {
     /// `APIClient` uses this object to decode the results of network requests.
     let decoder: RequestDecoder
     
+    /// Used for reobtating tokens when they expire and API client receives token expiration error
+    let tokenRefresher: (ClientError, @escaping () -> Void) -> Void
+    
     let cdnClient: CDNClient
+
+    /// Used for syncrhonizing access to requestsQueue
+    private let requestsAccessQueue = DispatchQueue(label: "io.getstream.requests")
+    
+    /// Stores request failed with token expired error for retrying them later
+    private var requestsQueue = [RequestsQueueItem]()
+    
+    /// Shows whether the token is being refreshed at the moment
+    private var isRefreshingToken: Bool = false
```

Comment:
unless access `isRefreshingToken` is serialized, I suggest using @Atomic to avoid surprises